internal/fsx/fsx_test.go

```diff
@@ -0,0 +1,74 @@
+package fsx_test
+
+import (
+	"bufio"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/ooni/probe-engine/internal/fsx"
+)
+
+var StateBaseDir = "../../testdata/fsx/"
+
+func TestFileInputInvalidFile(t *testing.T) {
+	dir, err := ioutil.TempDir(StateBaseDir, "test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+
+	// test function panics with invalid file
+	defer func() { recover() }()
+
+	dir += "/test.txt"
+	_, err = fsx.Open(dir)
+	if err == nil {
+		t.Errorf("expected panic with invalid input file")
+	}
+
+}
+
+func TestFileInputDirectory(t *testing.T) {
+	dir, err := ioutil.TempDir(StateBaseDir, "test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+
+	// test function panics with invalid directory
+	defer func() { recover() }()
+
+	_, err = fsx.Open(dir)
+	if err == nil {
+		t.Errorf("expected panic with input directory")
```

Comment:
```suggestion
		t.Fatalf("not the error we expected: %+v", err)
```

I think we also need to update the error string here.